my_enums.rb

```diff
@@ -1,60 +1,181 @@
 # frozen_string_literal: true
 
-# rubocop:disable Metrics/AbcSize, Metrics/CyclomaticComplexity, Metrics/MethodLength, Metrics/PerceivedComplexity
+# rubocop: disable Metrics/AbcSize
+# rubocop: disable Metrics/CyclomaticComplexity
+# rubocop: disable Metrics/MethodLength
+# rubocop: disable Metrics/PerceivedComplexity
 
 module Enumerable #:nodoc: all
   def my_each
+    return to_enum unless block_given?
+
     length.times do |n|
       current = self[n]
       yield(current)
     end
+    self
   end
 
   def my_each_with_index
-    length.times do |n|
-      index = n
-      current = self[n]
-      yield(index, current)
+    return to_enum unless block_given?
+
+    my_each do |n|
+      yield(n, index(n))
     end
+    self
   end
```

Comment:
should have working `my_each_with_index` **(covered by automated tests)** identical and returns the same thing as ruby's [`each_with_index`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-each_with_index). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_each_with_index.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
  - my_each_with_index calls the given block once for each element in self

The second argument of **yield** is supposed to be the index of the element in regards to the whole array. The method index that you are passing as a second argument is not giving the right index :pray: fix it